lib/logster/pattern.rb

```diff
@@ -1,7 +1,15 @@
+# frozen_string_literal: true
+
 module Logster
   class Pattern
+    ALL = []
     class PatternError < StandardError; end
 
+    def self.inherited(subclass)
+      Logster::Pattern::ALL << subclass
```

Comment:
ALL can be a little bit misleading because this is a constant, but dynamic :) What do you think of using a class variable?)